### ISSUE TITLE
Add support for wayland

### ISSUE THREAD

**Issue by LucaCoduriV** [ChurchTao/clipboard-rs#2]
It would be nice to make it compatble with wayland aswell.

**Comment by ChurchTao** [comment]
@LucaCoduriV  Yes, but this period is the Spring Festival. I'm going to add this later.

**Comment by getreu** [comment]
https://github.com/getreu/tp-note/blob/master/tpnote/src/clipboard.rs

**Comment by LucaCoduriV** [comment]
Hi, sorry what is this link ?

**Comment by getreu** [comment]
I combined your crate with wl-clipboard-rs. Maybe it helps to integrate Wayland support into your crate.
PS: I do not evaluate images.

**Comment by Ciubix8513** [comment]
The challenge with Wayland is, here is a protocol for doing clipboard stuff without a window,  but it is not supported in some major compositors.

The way youre supposed to handle it, is by going through a Wayland surface, there's a crate for that https://github.com/smithay/smithay-clipboard

**Comment by sincekmori** [comment]
clipboard-rs user here. My crate [copycopy](https://github.com/sincekmori/copycopy) uses it for multi-format reads on Windows and macOS (thanks for it!), and I ran into this same wall on Wayland. For GNOME in particular, where `wlr_data_control` isn't available as noted above, the only route I found was reading the clipboard from inside the compositor via a GNOME Shell extension. Not something clipboard-rs could adopt as-is, but the [extension source](https://github.com/sincekmori/copycopy/tree/main/src/gnome) might be useful as a reference for the GNOME side.
